Add Validate and Apply Operations Example (#1023)
mulesoft/docs-connectors: wss/1.0/modules/ROOT/pages/apply-wss-examples.adoc

= Apply Operation Examples

The `apply` operation applies WSS information to a SOAP response according to the configuration specified in `WssOutboundConfig`.

> .Parameters

Reviewer: Formatting:
We do not number or title tables like this; instead, provide and introductory text snippet that ends in a colon:
```suggestion
The operation has the following parameters:
```

Reviewer: Are the ".<text>" blue items that transform as titles supposed to be alt-text?
See ".Encryption Example" below.

Author: ![this](https://user-images.githubusercontent.com/16763154/92273234-5deaf700-eec1-11ea-9ef8-2e530a261fb6.png)

The . before a table turns into what's highlighted in the screenshot above.

[%header%autowidth.spread,cols="a*"]
|===
| Name | M | ES | Default Value | Type | Description
| `response` | Yes | No | - | `TypedValue <InputStream>` | SOAP response

> | `version` | No | Yes | `SOAP_12` | `SoapVersion` | SOAP version. Possible vlues: `SOAP_11`, `SOAP_12`

Reviewer: ```suggestion
| `version` | No | Yes | `SOAP_12` | `SoapVersion` | SOAP version: `SOAP_11` or `SOAP_12`
```

|===

> Output::

Reviewer: Format: neither do we use bold italic inline headings:
```suggestion
```

> The output is a SOAP response that contains the header element `<wsse:Security></wsse:Security>` resulting from applying the outbound configuration. The operation does not return any attribute.

Reviewer: ```suggestion
The output is a SOAP response that contains the header element `<wsse:Security></wsse:Security>`, resulting from applying the outbound configuration. The operation does not return any attributes.
```

> .Errors

Reviewer: ditto previous for .Parameters

[%header%autowidth.spread,cols="a*"]
|===
| Namespace | Type | Parent | Description

> | WSS | SECURITY_APPLYING | - | Thrown when failing to apply WSS in the response

Reviewer: ```suggestion
| WSS | SECURITY_APPLYING | n.a. | Thrown when failing to apply WSS in the response
```

> | WSS | MISSING_CERTIFICATE | - | Thrown when unable to get a certificate either from TrustStore or from Key Store

Reviewer: ```suggestion
| WSS | MISSING_CERTIFICATE | n.a. | Thrown when unable to get a certificate from either the trustStore or the keystore
```
Our Term Presentation List shows lowercase "truststore" and "keystore," all one word; please let me know if that is incorrect? If these are the syntax presentations, please use ` ` formatting.

|===

[[encrypt]]

> == Encrypt

Reviewer: This is not a Task topic and both narrative and syntax use "encryption," so...:
"Encryption Example"?
"Apply Encryption Example"?
GLOBAL for peer titles; I won't mark each one that follows.

> This example shows you how to apply encryption using a key-store:

Reviewer: ```suggestion
This example shows you how to apply encryption using a keystore:
```

> .Encyption Example

Reviewer: omit or code as alt-text

[source,XML,linenums]
----
<wss:outbound-config name="encryption-config">
	<wss:encryption-config>
		<wss:keystore-config path="certificates/sign-keystore.jks"
password="mulepassword"
alias="muleclient"
keyPassword="mulepassword" />
	</wss:encryption-config>
</wss:outbound-config>

<flow name="OrderTshirtServiceFlow">
	<http:listener config-ref="HTTP-listener-config" path="/order" />
	<flow-ref name="OrderTshirtFlowImpl" />
	<wss:apply-wss config-ref="encryption-config" version="SOAP_12"/>
</flow>
----

[[signature]]
== Sign

> This exampe shows you how to sign a SOAP response using key-store and the `RSAwithSHA1` algorithm:

Reviewer: ```suggestion
This exampe shows you how to sign a SOAP response using a keystore and the `RSAwithSHA1` algorithm:
```

.Signature Example
[source,XML,linenums]
----
<wss:outbound-config name="sign-config">
	<wss:signature-config algorithm="RSAwithSHA1" >
		<wss:keystore-config path="certificates/sign-keystore.jks"
                            	password="mulepassword"
					alias="muleclient"
       	                    keyPassword="mulepassword" />
	</wss:signature-config>
</wss:outbound-config>

<flow name="OrderTshirtServiceFlow">
	<http:listener config-ref="HTTP-listener-config" path="/order" />
	<flow-ref name="OrderTshirtFlowImpl" />
	<wss:apply-wss config-ref="sign-config" version="SOAP_12"/>
</flow>
----

[[encryption-signing-certificate]]

> == Encrypt And Sign a Certificate

Reviewer: ```suggestion
== Encrypt and Sign a Certificate
```
However...ditto prev. suggestion for all these titles; they are not Tasks.

> This example validates a signed request, saves the signing certificate and uses it later in the `apply-wss` operation to sign the response:

Reviewer: ```suggestion
This example validates a signed request, saves the signing certificate, and uses it later in the `apply-wss` operation to sign the response:
```

aopxUafPdlPkpfLeJOxMwggWTrjlszOoSq8ALLKybvhioBQYwV5Fm9rB04yalvbWCjCvQTaq83KH
WPpzKWfLXHXG4JV6m5Rue90ybfYHIM+FOgmC9CBbQX29wPz9pDCh1bGydEiSbyWje/frec0Jlw9M
vOCzPz5TbDJdOmVIKoSHTWPpeV3lZIG9vLYd4CYpw54CsZtc20QzWAF/DPlLC8TsL0rIgIxDCKYb
lFlAANVcl1CabqPziNBhep+9cB/oNHca3BVSQw==</ds:SignatureValue><ds:KeyInfo Id="KI-F8FAC4A91BEF76355615530303348132"><wsse:SecurityTokenReference wsu:Id="STR-F8FAC4A91BEF76355615530303348153"><ds:X509Data><ds:X509IssuerSerial><ds:X509IssuerName>CN=Unknown,OU=Unknown,O=Unknown,L=Unknown,ST=Unknown,C=US</ds:X509IssuerName><ds:X509SerialNumber>1545521240</ds:X509SerialNumber></ds:X509IssuerSerial></ds:X509Data></wsse:SecurityTokenReference></ds:KeyInfo></ds:Signature></wsse:Security></soapenv:Header>
   <soapenv:Body wsu:Id="id-F8FAC4A91BEF76355615530303348174" xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd">
      <ser:echo>
         <!--Optional:-->
         <text>test</text>
      </ser:echo>
   </soapenv:Body>
</soapenv:Envelope>
----

> 

Reviewer: Insert a bit of text here to transition from the previous syntax to this.
